compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -3053,7 +3054,8 @@ public BType getRestMatchPatternConstraintType(BRecordType recordType,
 
         if (!recordType.sealed) {
             BType restFieldType = recordType.restFieldType;
-            if (!this.types.isNeverTypeOrStructureTypeWithARequiredNeverMember(restFieldType)) {
+            if (!this.types.isNeverTypeOrStructureTypeWithARequiredNeverMember(restFieldType) &&
+                    (restFieldType.tag != TypeTags.NONE)) {
```

Comment:
IIRC, if the record type is closed (sealed) the rest field type is set to other (`TypeTags.NONE`), but an open record cannot have other as the rest type. What's the scenario here? Just wondering if the type isn't set correctly, in which case, this may not be the place to fix.

Reply:
Actually now due to the changes done in `createRecordTypeForRestField` and `setRestRecordFields` methods, we have correctly set the record type as sealed and hence we do not need this change here. Now, if the record is open, `BNoType` does not come as the rest type. So removed the change done here.